code/game/objects/items/airbubble.dm

```diff
@@ -0,0 +1,462 @@
+// Regular airbubble - folded
+/obj/item/airbubble
+	name = "air bubble"
+	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments."
+	icon = 'icons/obj/airbubble.dmi'
+	icon_state = "airbubble_fact_folded"
+	w_class = ITEMSIZE_NORMAL
+	var/used = FALSE
+	var/ripped = FALSE
+	var/obj/item/weapon/tank/internal_tank
+
+/obj/item/airbubble/Destroy()
+	qdel(internal_tank)
+	return ..()
+
+/obj/item/airbubble/attack_self(mob/user)
+	var/obj/structure/closet/airbubble/R = new /obj/structure/closet/airbubble(user.loc)
+	if(!used && !ripped)
+		internal_tank = new /obj/item/weapon/tank/emergency_oxygen/double(src)
+	R.internal_tank = internal_tank
+	internal_tank.forceMove(R)
+	internal_tank = null
+	R.add_fingerprint(user)
+	R.ripped = ripped
+	R.update_icon()
+	qdel(src)
+
+/obj/structure/closet/airbubble
+	name = "air bubble"
+	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments."
+	icon = 'icons/obj/airbubble.dmi'
+	icon_state = "airbubble"
+	icon_closed = "airbubble"
+	icon_opened = "airbubble_open"
+	open_sound = 'sound/items/zip.ogg'
+	close_sound = 'sound/items/zip.ogg'
+	var/item_path = /obj/item/airbubble
+	var/zipped = FALSE
+	density = 0
+	storage_capacity = 20
+	var/contains_body = FALSE
+	var/used = TRUE
+	var/ripped = FALSE
+
+	var/use_internal_tank = TRUE
+	var/datum/gas_mixture/inside_air
+	var/internal_tank_valve = 45 // arbitrary for now
+	var/obj/item/weapon/tank/internal_tank
+	var/process_ticks = 0
+
+/obj/structure/closet/airbubble/can_open()
+	if(zipped)
+		return 0
+	return 1
+
+/obj/structure/closet/airbubble/can_close()
+	if(zipped)
+		return 0
+	var/turf/T = get_turf(src)
+	for(var/obj/structure/closet/closet in T)
+		if(closet != src)
+			return 0
+	var/mob_num = 0
+	for(var/mob/living/M in T)
+		mob_num += 1
+		if(mob_num > 1)
+			visible_message("<span class='warning'>[src] can only fit one person.</span>")
+			return 0
+	return 1
+
+/obj/structure/closet/airbubble/dump_contents()
+
+	for(var/obj/I in src)
+		if(!istype(I, /obj/item/weapon/tank))
```

Comment:
Bad idea, you might legitimately have multiple air tanks in the `src`. Do this instead:

```
if (I != internal_tank)
    I.forceMove(loc)
```